### PR TITLE
fix rvs build issue by using yaml-cpp spack recipe

### PR DESCRIPTION
**Pull request by srekolam**
fix the build issue that is seen with rvs #24430 ,
modify the rocfft cmake args for  #25773

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
LGTM

File: var/spack/repos/builtin/packages/rocm-validation-suite/package.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ class RocmValidationSuite(CMakePackage):
 
     depends_on('cmake@3.5:', type='build')
     depends_on('zlib', type='link')
+    depends_on('yaml-cpp', type='build')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rocm-validation-suite/package.py]: Why type=build? Doesn't look right to me. The cmake script seems to download/build yaml-cpp itself though? Can that be disabled so that we can use spack's yaml-cpp?

**Review by haampie — CHANGES_REQUESTED** [review]
See the above comment

File: var/spack/repos/builtin/packages/rocfft/package.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def cmake_args(self):
         # From version 3.9 and above we have AMDGPU_TARGETS_SRAM_ECC
         tgt_sram = self.spec.variants['amdgpu_target_sram_ecc'].value
 
-        if tgt_sram[0] != 'none' and '@3.9.0:' in self.spec:
+        if tgt_sram[0] != 'none' and self.spec.satisfies('@3.9.0:4.0.0'):
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rocfft/package.py]: This is unrelated to the PR? Can you open a separate PR for that?

Comment (srekolam) [reply var/spack/repos/builtin/packages/rocfft/package.py]: okay.. i have reverted the rocfft changes and will open a separate PR

**Comment by srekolam** [comment]
> See the above comment

done..

**Review by tldahlgren — DISMISSED** [review]

**Comment by tldahlgren** [comment]
@haampie Looks like you need to re-review this PR in order for it to be merged.

File: var/spack/repos/builtin/packages/rocm-validation-suite/package.py [hunk 3]
```diff
@@ -55,5 +57,7 @@ def cmake_args(self):
         return [
             self.define('HIP_INC_DIR', self.spec['hip'].prefix),
             self.define('ROCM_SMI_DIR', self.spec['rocm-smi-lib'].prefix),
-            self.define('ROCBLAS_DIR', self.spec['rocblas'].prefix)
+            self.define('ROCBLAS_DIR', self.spec['rocblas'].prefix),
+            self.define('YAML_INC_DIR', self.spec['yaml-cpp'].prefix.include),
+            self.define('YAML_LIB_DIR', self.spec['yaml-cpp'].prefix.lib64)
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rocm-validation-suite/package.py]: lib64 is not guaranteed to be the right directory, see other packages that use `libs.directories`

**Review by haampie — CHANGES_REQUESTED** [review]

**Comment by haampie** [comment]
I'm inclined to say that this package is just hopelessly unpackage-able. 

They should allow the user to disable vendored dependencies, and the yaml library is just one of the many:

```
CMakeRBLASDownload.cmake
CMakeRSMIDownload.cmake
CMakeYamlDownload.cmake
```

I'd rather not have yet another patch, but see if @srekolam can convince other AMD folks to get rid of the forced download of dependencies during the cmake phase.

**Comment by haampie** [comment]
@srekolam: https://github.com/ROCm-Developer-Tools/ROCmValidationSuite/issues/502

**Comment by srekolam** [comment]
> I'm inclined to say that this package is just hopelessly unpackage-able.
> 
> They should allow the user to disable vendored dependencies, and the yaml library is just one of the many:
> 
> ```
> CMakeRBLASDownload.cmake
> CMakeRSMIDownload.cmake
> CMakeYamlDownload.cmake
> ```
> 
> I'd rather not have yet another patch, but see if @srekolam can convince other AMD folks to get rid of the forced download of dependencies during the cmake phase.

i have asked the internal team to  look at your ROCm-Developer-Tools/ROCmValidationSuite#502 . Will follow up with the team for cleanup. 
Can we take this fix in the interim ?

**Comment by srekolam** [comment]
@haampie , can you please let me know if this can be merged. The internal team is working on the issues that you have raised .This will be done for future release.

**Review by haampie — APPROVED** [review]
